distribution/src/bin/elasticsearch-env

```diff
@@ -35,6 +35,31 @@ ES_HOME=`dirname "$ES_HOME"`
 # now set the classpath
 ES_CLASSPATH="$ES_HOME/lib/*"
 
+
+check_glibc_version() {
+  # We need to check the version of glibc, because JDK >=15 requires
+  # glibc 2.14
+  GLIBC_VERSION="$( getconf GNU_LIBC_VERSION 2>/dev/null | cut -d' ' -f2)"
+  if [[ $? -ne 0 ]]; then
```

Comment:
Does this work? I thought the `$?` would be the result of the assignment, not the inner command?

Reply:
`$?` is the status of the most recent pipeline, and variable assignment is not a pipeline.
So, with respect to the assignment it works. e.g.
```
bash -c 'False=$( false ) ; echo $? ; True=$(true) ; echo $?'
1
0
```

However, it only works because the script sets `pipefail`, otherwise the status of `getconf` would be lost and we would be testing the status of `cut` (which would succeed).

_Back to lurk mode.  
Sorry I spent years maintaining a system that had in excess of 10k lines of ksh code, some of this stuff is burned into my brain_.

Reply:
Thanks Tim! I've reworked the script to be more cautious, and also to stop performing floating point comparisons.